controller/konnect/konnectextension_controller_utils_test.go

```diff
@@ -0,0 +1,123 @@
+package konnect
+
+import (
+	"testing"
+
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+)
+
+func TestEnforceKonnectExtensionStatus(t *testing.T) {
+	cp := konnectv1alpha1.KonnectGatewayControlPlane{
+		Status: konnectv1alpha1.KonnectGatewayControlPlaneStatus{
+			KonnectEntityStatus: konnectv1alpha1.KonnectEntityStatus{
+				ID: "cp-id",
+			},
+			Endpoints: &konnectv1alpha1.KonnectEndpoints{
+				ControlPlaneEndpoint: "cp-endpoint",
+				TelemetryEndpoint:    "telemetry-endpoint",
+			},
+		},
+		Spec: konnectv1alpha1.KonnectGatewayControlPlaneSpec{
+			CreateControlPlaneRequest: konnectv1alpha1.CreateControlPlaneRequest{
+				ClusterType: sdkkonnectcomp.CreateControlPlaneRequestClusterTypeClusterTypeControlPlane.ToPointer(),
+			},
+		},
+	}
+	certificateSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "my-secret",
+		},
+	}
+
+	t.Run("updates both Konnect and DataPlaneClientAuth when both are different", func(t *testing.T) {
+		ext := &konnectv1alpha1.KonnectExtension{
+			Status: konnectv1alpha1.KonnectExtensionStatus{
+				Konnect:             nil,
+				DataPlaneClientAuth: nil,
+			},
+		}
+		updated := enforceKonnectExtensionStatus(cp, certificateSecret, ext)
+		assert.True(t, updated)
+		assert.NotNil(t, ext.Status.Konnect)
```

Comment:
To prevent panics below when sub fields of `ext.Status.Konnect` are accessed.

```suggestion
		require.NotNil(t, ext.Status.Konnect)
```